File: api/models/Characteristic.js
module.exports = (sequelize, DataTypes) => {
  const Characteristic = sequelize.define(
    "Characteristic",
    {
      id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
      },
      species_id: {
        type: DataTypes.INTEGER,
        allowNull: false,
        references: {
          model: "Species",
          key: "id",
        },
      },
      characteristic_type: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      value: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      units: {
        type: DataTypes.STRING,
        allowNull: true,
      },
    },
    {
      tableName: "Characteristic",
      timestamps: false,
    },
  );

  return Characteristic;
};

File: app/lib/models/Characteristic.dart
// model/characteristic.dart
class Characteristic {
  final int id;
  final String type;
  final String value;
  final String units;

  Characteristic({
    required this.id,
    required this.type,
    required this.value,
    required this.units,
  });

  factory Characteristic.fromJson(Map<String, dynamic> json) {
    return Characteristic(
      id: json['id'],
      type: json['characteristic_type'],
      value: json['value'],
      units: json['units'],
    );
  }
}
